### PR TITLE
impl config server

### PR DESCRIPTION
**Pull request by hr3lxphr6j**

### CODE REVIEW

**Comment by hr3lxphr6j** [comment]
@kirk91 PING

File: discovery/dependency.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		del = s.lastDeps
+	}
+	s.lastDeps = deps
+	return &api.DependencyDiscoveryResponse{
+		Added:   buildServices(add...),
+		Removed: buildServices(del...),
+	}, nil
+}
+
+func (s *dependencyDiscoverySession) SetUnsubscribeHandler(hdlr dependencyUnsubHandler) {
```
Comment (kirk91) [inline discovery/dependency.go]: Why need unsubscribe handler?  `DependencyDiscoverySession` doesn't subscribe any services.

File: discovery/dependency.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	quit chan struct{}
+}
+
+func newDependencyDiscoverySession(instID string, stream api.DiscoveryService_StreamDependenciesServer) *dependencyDiscoverySession {
+	remote, _ := peer.FromContext(stream.Context())
+	return &dependencyDiscoverySession{
+		instID:  instID,
+		stream:  stream,
+		remote:  remote,
+		eventCh: make(chan *config.Event, 64),
```
Comment (kirk91) [inline discovery/dependency.go]: the capacity size 64 is a little big

File: config/controller.go [hunk 3]
```diff
@@ -316,32 +309,22 @@ func (c *Controller) RegisterEventHandler(handler EventHandler) {
 
 // Get return config data by namespace, type and key.
 func (c *Controller) Get(namespace, typ, key string) ([]byte, error) {
+	return c.store.Get(namespace, typ, key)
+}
+
+// Get return config data by namespace, type and key from cache.
```
Comment (kirk91) [inline config/controller.go]: nit: `GetCache`

**Pull request merged**